blockio-sim/src/System/FS/BlockIO/Sim.hs

```diff
@@ -21,7 +21,9 @@ fromHasFS ::
      (MonadThrow m, MonadMVar m, PrimMonad m)
   => HasFS m HandleMock
   -> m (HasBlockIO m HandleMock)
-fromHasFS = serialHasBlockIO
+fromHasFS = serialHasBlockIO hSetNoCache
+  where
+    hSetNoCache _h _b = pure ()
```

Comment:
I'm on the fence about whether the implementations and simulation should throw an `FsError` when subsequently doing file I/O with misaligned byte arrays... Maybe they should? However, to do that we'd have to be able to move `hSetNoCache` into `fs-api` and `fs-sim`...

Reply:
That's a nice improvement to be able to make later.